common/builder/tests/test_pystache_message_builder.py

```diff
@@ -19,3 +19,7 @@ def test_build_message(self):
 
         self.assertEqual("Hello, world!", message.strip(),
                          "Message returned should be the rendered string returned by Pystache")
+
+    def test_build_message_errors_on_missing_tag(self):
+        with self.assertRaises(MessageGenerationSOAPFaultError):
+            self.builder.build_message({})
```

Comment:
If I'm thinking about this correctly:
If we want the optional tags to default to False then it seems that this would have to be done at a point where it is known what the template is (so that we know what the optional tags are), and then set defaults for those tags specifically. I haven't done this, I think we only need to do this if we find that there's a decent chunk of time we're spending specifying these default tags.

Reply:
Seems reasonable to leave it as it is, like you say we can always rethink it if we're defaulting a lot of tags